src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -328,6 +332,15 @@ private void handleFetchingException(ExecutionContext executionContext,
         Object result = unboxPossibleOptional(parameters.source());
         GraphQLType fieldType = typeInfo.getType();
 
+        if (result != null && result instanceof DataFetcherResult) {
```

Comment:
Wasn't sure if this should be better integrated into the if/else section below or there was value in keeping it all separate from a readability standpoint

Reply:
I don't think this is the right place: it should be handled directly when the DataFetcher result is returned.

Reply:
I did move it over to fetchField, which seems to work well.